sqrl-planner/sqrl-planner-local/src/main/java/com/datasqrl/frontend/SqrlPlan.java

```diff
@@ -1,48 +1,54 @@
 package com.datasqrl.frontend;
 
+import com.datasqrl.canonicalizer.NameCanonicalizer;
 import com.datasqrl.error.ErrorCollector;
 import com.datasqrl.loaders.ModuleLoader;
-import com.datasqrl.canonicalizer.NameCanonicalizer;
+import com.datasqrl.loaders.ModuleLoaderComposite;
 import com.datasqrl.parse.SqrlParser;
-import com.datasqrl.plan.local.generate.Debugger;
-import com.datasqrl.plan.local.generate.DebuggerConfig;
-import com.datasqrl.plan.local.generate.Namespace;
-import com.datasqrl.plan.local.generate.NamespaceFactory;
-import com.datasqrl.plan.local.generate.Resolve;
-import com.datasqrl.plan.local.generate.SqrlQueryPlanner;
-import com.datasqrl.plan.local.generate.StatementProcessor;
+import com.datasqrl.plan.local.generate.*;
+import com.datasqrl.plan.table.CalciteTableFactory;
 import com.google.inject.Inject;
 import org.apache.calcite.sql.ScriptNode;
 
+import java.util.List;
+
 public class SqrlPlan extends SqrlParse {
 
   protected NamespaceFactory nsFactory;
   protected ModuleLoader moduleLoader;
   protected NameCanonicalizer nameCanonicalizer;
-  protected StatementProcessor statementProcessor;
+  protected CalciteTableFactory tableFactory;
   protected SqrlQueryPlanner planner;
   private final DebuggerConfig debuggerConfig;
 
   @Inject
   public SqrlPlan(SqrlParser parser, ErrorCollector errors, NamespaceFactory nsFactory,
-      ModuleLoader moduleLoader, NameCanonicalizer nameCanonicalizer,
-      StatementProcessor statementProcessor, SqrlQueryPlanner planner, DebuggerConfig debuggerConfig) {
+      ModuleLoader moduleLoader, NameCanonicalizer nameCanonicalizer, CalciteTableFactory tableFactory,
+      SqrlQueryPlanner planner, DebuggerConfig debuggerConfig) {
     super(parser, errors);
     this.nsFactory = nsFactory;
     this.moduleLoader = moduleLoader;
     this.nameCanonicalizer = nameCanonicalizer;
-    this.statementProcessor = statementProcessor;
+    this.tableFactory = tableFactory;
     this.planner = planner;
     this.debuggerConfig = debuggerConfig;
   }
 
-  public Namespace plan(String script) {
-    return plan(parse(script));
+  public Namespace plan(String script, List<ModuleLoader> additionalModules) {
+    return plan(parse(script), additionalModules);
   }
 
-  public Namespace plan(ScriptNode node) {
-    Resolve resolve = new Resolve(this.nsFactory, this.moduleLoader,
-        this.nameCanonicalizer, this.errors, this.statementProcessor);
+  public Namespace plan(ScriptNode node, List<ModuleLoader> additionalModules) {
+    ModuleLoader updatedModuleLoader = this.moduleLoader;
+    if (!additionalModules.isEmpty()) {
+      updatedModuleLoader = ModuleLoaderComposite.builder().
+          moduleLoader(this.moduleLoader).
+          moduleLoaders(additionalModules).
```

Comment:
The chain of responsibility happens in the ModuleLoader, not here. It will make it hard to wrangle the correct construction of the module loader if it's spread out like this.

Also, Dots (.) should be at the start.

Reply:
Not sure what you mean by this. ModuleLoader is just an interface.

Reply:
If you want to go this route of abstracting module loaders, then the other logic in ModuleLoaderImpl should also be converted